pkg/loop/internal/core/services/keyvaluestore/keyvalue_store.go

```diff
@@ -0,0 +1,91 @@
+package keyvaluestore
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/loop/internal/pb"
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+)
+
+var _ types.KeyValueStore = (*Client)(nil)
+
+type Client struct {
+	grpc pb.KeyValueStoreClient
+}
+
+func (k Client) Store(ctx context.Context, key string, val any) error {
+	jsonVal, err := json.Marshal(val)
```

Comment:
@ettec Hmmm... this doesn't seem quite right as I'm sure you've figured out :) Could we instead make the underlying service just store arbitrary bytes --potentially with some validation that the bytes are valid JSON? That way the GRPC layer would be relatively simple and could just pass the bytes along.

Reply:
right, I was a little unsure about that, and whether the type passed in should maintain the interface as outlined in the orm_kv.go. or just have the bytes methods.  Question -  would it not be reasonable for the client of the store to want to store some arbitrary bytes, and perhaps we add a convenience method for the the case where they want to serialise the object via json?